src/pdm/cli/commands/list.py

```diff
@@ -234,7 +234,7 @@ def _group_of(name: str) -> set[str]:
             keys = parse_comma_separated_string(options.sort)
             if not all(key in Listable.KEYS for key in keys):
                 raise PdmUsageError(f"--sort key must be one of: {','.join(Listable.KEYS)}")
-            records.sort(key=lambda d: tuple(d[key] for key in keys))
+            records.sort(key=lambda d: tuple(d[key].lower() for key in keys))
```

Comment:
if it was just package name, lower should be fine however since there are other fields, maybe the slightly more unicode correct `.casefold()` can be used instead for the key, that's its purpose.